src/Microsoft.AspNetCore.Sockets/Internal/ConnectionState.cs

```diff
@@ -75,6 +80,9 @@ public ConnectionState(Connection connection, IChannelConnection<Message> applic
 
             // REVIEW: Add a timeout so we don't wait forever
             await Task.WhenAll(applicationTask, transportTask);
```

Comment:
Couldn't we just move the `Task.WhenAll` inside the `Lock` and it would achieve the same thing without creating a tcs?